client/blocks/inline-help/popover.jsx

```diff
@@ -168,18 +199,51 @@ class InlineHelpPopover extends Component {
 }
 
 function mapStateToProps( state ) {
+	const siteId = getSelectedSiteId( state );
+	const currentRoute = getCurrentRoute( state );
+	const classicRoute = currentRoute.replace( '/gutenberg/', '' );
+
 	return {
 		searchQuery: getSearchQuery( state ),
 		selectedSite: getHelpSelectedSite( state ),
 		selectedResult: getInlineHelpCurrentlySelectedResult( state ),
+		selectedEditor: getSelectedEditor( state, siteId ),
+		classicUrl: `/${ classicRoute }`,
+		siteId,
 	};
 }
 
-export default connect(
-	mapStateToProps,
-	{
+function mapDispatchToProps( dispatch ) {
+	return {
+		optOut: siteId => {
+			dispatch(
+				withAnalytics(
+					composeAnalytics(
+						recordGoogleEvent(
+							'Gutenberg Opt-Out',
+							'Clicked "Switch to the classic editor" in the help popover.',
+							'Opt-In',
+							false
+						),
+						recordTracksEvent( 'calypso_gutenberg_opt_in', {
+							opt_in: false,
+						} ),
+						bumpStat( 'gutenberg-opt-in', 'Calypso Help Opt Out' )
+					),
+					setSelectedEditor( siteId, 'classic' )
+				)
+			);
+		},
```

Comment:
Consider pulling this out to it's own function:
```jsx
const optOut = ( siteId ) => {
    return  withAnalytics(
					composeAnalytics(
						recordGoogleEvent(
							'Gutenberg Opt-Out',
							'Clicked "Switch to the classic editor" in the help popover.',
							'Opt-In',
							false
						),
						recordTracksEvent( 'calypso_gutenberg_opt_in', {
							opt_in: false,
						} ),
						bumpStat( 'gutenberg-opt-in', 'Calypso Help Opt Out' )
					),
					setSelectedEditor( siteId, 'classic' )
         );
}

const redirect = ( classicUrl ) => {
    return navigate( classicUrl );
}
```

And then we can keep object notation:
```jsx
const mapDispatchToProps = {
    optOut,
    recordTracksEvent,
    //...
    redirect,
};
```

Reply:
Much cleaner too, I'll adjust!

Reply:
Updated in 179f821.